### DISCUSSION TITLE
Numeric data component

### DISCUSSION THREAD

**Discussion by jenthomson** [ministryofjustice/moj-frontend#290] [EXPERIMENTAL: Components, pages and patterns]
![image](https://user-images.githubusercontent.com/1866734/147359426-76571618-1099-454f-a421-50f956a8a2a9.png)

## Existing

Similar published components or patterns
https://digital-land-design.herokuapp.com/component/data-item

GitHub discussions on similar topics
https://github.com/ministryofjustice/moj-frontend/discussions/210

## When to use
Use the numeric data pattern to make pieces of information easier to scan, interpret and action where required.

### Writing content
The number part of the component can be written as an integer, a decimal, an amount of currency or a percentage. 

It should not be presented as a word or a fraction.

The label should be short and succinct. Recommend no more than 20 characters.

### Making sure it’s accessible

- Recommended use is black text, on white, or gov-uk light-grey, background.
- Labels and text should make sense when you read it visually or with screenreader
- If using multiple instances:
-- use a heading to provide context
-- check for responsiveness on multiple screen sizes
- If using a link, check that it makes sense to a screen reader
- 

### Implementing the code
Use spans to wrap the elements 
The numeric part should be maximum 48px (govuk-heading-xl) or minimum 36px (govuk-heading-l)
The label should be normal weight, maximum 24px (govuk-body-l) or minimum 19px (govuk-body)
For the first iteration, labels should sit underneath the number
Text should be aligned left
Links can be used to take a user to the detail, but preferably link the label.

Using it in context with other components and as part of a service?
Use flex-box to display multiple instances in one row

### When to use variations? (if applicable)

First iteration will have only one variation: Data stacked on top of label, aligned left.

## Things we don’t know enough about

- Linking to the detail or filtering data
- Different layout, size or colour combinations

**Comment by helennickols** [comment]
The numeric data component has been published on the MOJ Design System. This is to make it easier to find and use.

https://design-patterns.service.justice.gov.uk/components/numeric-data/

**Reply by mseedat-moj** [reply]
Is there any plan to publish example HTML code for this component? Would be helpful on a dashboard I am building. Thanks

**Reply by helennickols** [reply]
@mseedat-moj Hi Mo, I've asked the PM on the project and will let you know. Thanks!

**Reply by helennickols** [reply]
@mseedat-moj I've sent you the code by Slack DM. We'll also add it to the documentation.
